lxd-qemu-snap/README.md

```diff
@@ -0,0 +1,72 @@
+# External QEMU snap for LXD snap
+
+## How to build and use
+
+```
+cd lxd-qemu-snap
+
+# clean up previous builds
+rm qemu-for-lxd_*.snap
+
+# build
+snapcraft
+
+# install
+sudo snap install qemu-for-lxd_*.snap --devmode
+
+# connect snaps
+sudo snap connect lxd:external-qemu qemu-for-lxd:external-qemu
+sudo snap connect lxd:gpu-2404 mesa-2404:gpu-2404
+
+# make LXD to use QEMU from external snap
+sudo snap set lxd qemu.external-snap=true
+```
+
+## How to use with virgl (only specific to this example of snapcraft.yaml)
+
+```
+lxc init images:ubuntu/noble/desktop desktop -c limits.memory=8GiB --vm
+
+# modify instance configuration:
+lxc config edit desktop
+
+# choose an appropriate renderer:
+ls -la /dev/dri/by-path/*-render
+
+# for example, on my system it is /dev/dri/by-path/pci-0000:67:00.0-render
+# lspci | grep 67:00 shows:
+# 67:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Rembrandt (rev d8)
+
+# add the following lines:
+  raw.apparmor: |-
+    /snap/lxd/*/gpu-2404/** mr,
+    /dev/dri/{,**} rw,
+    /dev/dri/card[0-9]* rw,
+    /dev/dri/renderD[0-9]* rw,
```

Comment:
Aren't those 2 rules already covered by the `/dev/dri/{,**} rw,` one before?

Reply:
sure, let's do even better:
```
    /snap/lxd/*/gpu-2404/** mr,
    /dev/dri/ r,
    /dev/dri/card[0-9]* rw,
    /dev/dri/renderD[0-9]* rw,
    /run/udev/data/c226:[0-9]* r,  # 226 drm
    /sys/devices/** r,
    /sys/bus/** r,
```